test/integration/extensions_test.go

```diff
@@ -0,0 +1,229 @@
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+	}
+
+	// Create extension scripts.
+	var scripts []string
+	scriptFile1 := filepath.Join(tmpdir, "script1.js")
+	check(ioutil.WriteFile(scriptFile1, []byte(script1), 0644))
+	scripts = append(scripts, scriptFile1)
+
+	scriptFile2 := filepath.Join(tmpdir, "script2.js")
+	check(ioutil.WriteFile(scriptFile2, []byte(script2), 0644))
+	scripts = append(scripts, scriptFile2)
+
+	// Create extension stylesheets.
+	var stylesheets []string
+	stylesheetFile1 := filepath.Join(tmpdir, "stylesheet1.css")
+	check(ioutil.WriteFile(stylesheetFile1, []byte(stylesheet1), 0644))
+	stylesheets = append(stylesheets, stylesheetFile1)
+
+	stylesheetFile2 := filepath.Join(tmpdir, "stylesheet2.css")
+	check(ioutil.WriteFile(stylesheetFile2, []byte(stylesheet2), 0644))
+	stylesheets = append(stylesheets, stylesheetFile2)
+
+	// Create directories for other extension content.
+	dir1 := filepath.Join(tmpdir, "extension1")
+	check(os.Mkdir(dir1, 0755))
+	dir2 := filepath.Join(tmpdir, "extension2")
+	check(os.Mkdir(dir2, 0755))
+
+	// Add an index.html to both directories.
+	check(ioutil.WriteFile(filepath.Join(dir1, "index.html"), []byte(indexHTML), 0644))
+	check(ioutil.WriteFile(filepath.Join(dir2, "index.html"), []byte(indexHTML), 0644))
+
+	// Add a text file to a subfolder of dir1.
+	sudir := filepath.Join(dir1, "files")
+	check(os.Mkdir(sudir, 0755))
+	check(ioutil.WriteFile(filepath.Join(sudir, "shakespeare.txt"), []byte(plaintext), 0644))
+
+	// Build master config.
+	masterOptions, err := testutil.DefaultMasterOptions()
+	check(err)
+	masterOptions.AssetConfig.ExtensionScripts = scripts
+	masterOptions.AssetConfig.ExtensionStylesheets = stylesheets
+	masterOptions.AssetConfig.Extensions = []configapi.AssetExtensionsConfig{
+		{
+			Name:            "extension1",
+			SourceDirectory: dir1,
+			HTML5Mode:       true,
+		},
+		{
+			Name:            "extension2",
+			SourceDirectory: dir2,
+			HTML5Mode:       false,
+		},
+	}
+
+	return masterOptions
+}
+
+func concat(separator string, content ...string) []byte {
+	var buffer bytes.Buffer
+	for _, s := range content {
+		buffer.WriteString(s)
+		buffer.WriteString(separator)
+	}
+
+	return buffer.Bytes()
+}
+
+func injectBase(content, extensionName string, publicURL *url.URL) []byte {
+	base := path.Join(publicURL.Path, "extensions", extensionName) + "/"
+	return bytes.Replace([]byte(content), []byte(`<base href="/">`), []byte(fmt.Sprintf(`<base href="%s">`, base)), 1)
+}
+
+func testGET(t *testing.T, url string, expectedStatus int, expectedType string, expectedContent []byte) {
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		t.Errorf("Unexpected error creating request for %q: %v", url, err)
+	}
+
+	resp, err := transport.RoundTrip(req)
+	if err != nil {
+		t.Errorf("Unexpected error while accessing %q: %v", url, err)
```

Comment:
probably need to short circuit on some of these... Errorf doesn't stop the test, which means you can get nil panics if the test continues and tried to use a nil pointer